android/src/main/java/com/rnldk/RnLdkModule.kt

```diff
@@ -412,11 +415,23 @@ class RnLdkModule(private val reactContext: ReactApplicationContext) : ReactCont
 
     if (event is Event.PaymentReceived) {
       println("ReactNativeLDK: " + "payment received, payment_hash: " + byteArrayToHex(event.payment_hash));
-      channel_manager?.claim_funds(event.payment_preimage);
+      var paymentPreimage: ByteArray? = null;
+      var paymentSecret: ByteArray? = null;
+
+      if (event.purpose is PaymentPurpose.InvoicePayment) {
+        paymentPreimage = (event.purpose as PaymentPurpose.InvoicePayment).payment_preimage;
+        paymentSecret = (event.purpose as PaymentPurpose.InvoicePayment).payment_secret;
+        channel_manager?.claim_funds(paymentPreimage);
+      }
```

Comment:
You need an else block here - either claim keysend payments or fail them, but don't ignore them.